### PR TITLE
Clean up commissioner state names.

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: src/core/meshcop/dataset_manager.cpp [hunk 1]
```diff
@@ -601,11 +601,8 @@ otError DatasetManager::SendSetRequest(const otOperationalDataset &aDataset, con
     VerifyOrExit((message = NewMeshCoPMessage(mNetif.GetCoap(), header)) != NULL, error = OT_ERROR_NO_BUFS);
 
 #if OPENTHREAD_ENABLE_COMMISSIONER && OPENTHREAD_FTD
-    bool isCommissioner;
 
-    isCommissioner = mNetif.GetCommissioner().GetState() != kCommissionerStateDisabled ? true : false;
-
-    if (isCommissioner)
+    if (mNetif.GetCommissioner().IsActive())
```
Comment (abtink) [inline src/core/meshcop/dataset_manager.cpp]: :)

Comment (abtink) [inline src/core/meshcop/dataset_manager.cpp]: @jwhui, the earlier code was checking if we state is anything other than ` Disabled` to set `isCommisioner`...New code is checking if state is exactly `ACTIVE` (so it blocks if we get here with state being `PETITION`)...

I assume the new code is what we want and the intended behavior.... Correct?

Comment (jwhui) [inline src/core/meshcop/dataset_manager.cpp]: That was intentional.  A Commissioner does not have a valid Session ID until it is in the Active state.

Comment (jwhui) [inline src/core/meshcop/dataset_manager.cpp]: @xiaom-GitHub, what do you think of this change?

Comment (xiaom-GitHub) [inline src/core/meshcop/dataset_manager.cpp]: Agree. Only active commissioner will send a MGMT_ACTIVE_SET.req message to change one or more Active Operational Dataset values in the Leader. Here, for some test case, Harness intends to assign an invalid session id to verify if Leader can identify this invalid session id or not.

**Pull request merged**